src/language-js/printer-estree.js

```diff
@@ -5750,7 +5750,11 @@ function printBinaryishExpressions(
 
 function printAssignmentRight(leftNode, rightNode, printedRight, options) {
   if (hasLeadingOwnLineComment(options.originalText, rightNode, options)) {
-    return indent(concat([hardline, printedRight]));
+    const shouldPrintLine =
+      rightNode.comments &&
+      Array.isArray(rightNode.comments) &&
```

Comment:
These two line are redundant. We're already inside `if (hasLeadingOwnLineComment(...))`.